File: app/models/bike.rb
# == Schema Information
#
# Table name: bikes
#
#  id                :integer          not null, primary key
#  displacement      :integer
#  manufactured_date :datetime
#  name              :string
#  price             :float
#  created_at        :datetime         not null
#  updated_at        :datetime         not null
#  manufacturer_id   :integer
#
# Indexes
#
#  index_bikes_on_manufactured_date  (manufactured_date)
#  index_bikes_on_name               (name) UNIQUE
#
class Bike < ApplicationRecord
  belongs_to :manufacturer
end

File: frontend/models/bike.ts
import { toCamel } from "../lib/transform_keys";

export interface IBike {
  id: number,
  displacement: number,
  manufactured_date: Date,
  name: string,
  price: number,
  createdAt: Date,
  updatedAt: Date,
  manufacturerId: number,
}

export class Bike implements IBike {
  static readonly TYPE = 'bikes';

  id: number;
  displacement: number;
  manufactured_date: Date;
  name: string;
  price: number;
  createdAt: Date;
  updatedAt: Date;
  manufacturerId: number;

  constructor(jsonData: any) {
    const {
      id,
      displacement,
      manufactured_date,
      name,
      price,
      createdAt,
      updatedAt,
      manufacturerId
    } = jsonData;

    this.id = id;
    this.displacement = displacement;
    this.manufactured_date = manufactured_date;
    this.name = toCamel(name);
    this.price = price;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.manufacturerId = manufacturerId;
  }
}
